palantir-java-format/src/main/java/com/palantir/javaformat/doc/Level.java

```diff
@@ -122,17 +102,46 @@ String computeFlat() {
 
     @Override
     public State computeBreaks(CommentsHelper commentsHelper, int maxWidth, State state) {
-        float thisWidth = getWidth();
-        if (state.column() + thisWidth <= maxWidth) {
-            return state.withColumn(state.column() + (int) thisWidth)
-                    .withLevelState(this, ImmutableLevelState.of(true));
+        Optional<State> oneLine = tryToFitOnOneLine(maxWidth, state);
```

Comment:
```suggestion
        return tryToFitOnOneLine(maxWidth, state)
            .orElseGet(() -> state.updateAfterLevel(getBreakBehaviour().match(new BreakImpl(commentsHelper, maxWidth, state));
```

Reply:
yep looks better, tho I kept the orElse bit on two lines as I think it looks cleaner